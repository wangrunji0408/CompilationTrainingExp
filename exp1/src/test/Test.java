package test;

public class Test {
    public static void main (String [] args) {
	int a, b, c;
	a = 3;
	if (a < 2) {
	    b = 3;
	    c = 5;
	} else {
	    b = 3;
	    c = 6;
	}
	a = -a;
	c = a+b;
	System.out.println(a+" "+b+" "+c);
    }
}
