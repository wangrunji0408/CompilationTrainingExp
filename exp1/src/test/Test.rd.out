Now analyzing test.Test
Method: main
entry: []
1 in:  []
1 out: [1]
2 in:  [1]
2 out: [1]
3 in:  [1]
3 out: [1, 3]
4 in:  [1, 3]
4 out: [1, 3, 4]
5 in:  [1]
5 out: [1, 5]
6 in:  [1, 5]
6 out: [1, 5, 6]
7 in:  [1, 5, 6]
7 out: [1, 5, 6]
8 in:  [1, 3, 4, 5, 6]
8 out: [1, 3, 4, 5, 6, 8]
9 in:  [1, 3, 4, 5, 6, 8]
9 out: [3, 4, 5, 6, 8, 9]
10 in:  [3, 4, 5, 6, 8, 9]
10 out: [3, 4, 5, 6, 9, 10]
11 in:  [3, 4, 5, 6, 9, 10]
11 out: [3, 5, 9, 10, 11]
12 in:  [3, 5, 9, 10, 11]
12 out: [3, 5, 9, 10, 11, 12]
13 in:  [3, 5, 9, 10, 11, 12]
13 out: [3, 5, 9, 10, 11, 12, 13]
14 in:  [3, 5, 9, 10, 11, 12, 13]
14 out: [3, 5, 9, 10, 11, 12, 13, 14]
15 in:  [3, 5, 9, 10, 11, 12, 13, 14, 16]
15 out: [3, 5, 9, 10, 11, 12, 13, 14, 16]
16 in:  [3, 5, 9, 10, 11, 12, 13, 14]
16 out: [3, 5, 9, 10, 11, 12, 13, 14, 16]
17 in:  [3, 5, 9, 10, 11, 12, 13, 14, 18]
17 out: [3, 5, 9, 10, 11, 12, 14, 17, 18]
18 in:  [3, 5, 9, 10, 11, 12, 13, 14, 16]
18 out: [3, 5, 9, 10, 11, 12, 13, 14, 18]
19 in:  [3, 5, 9, 10, 11, 12, 17, 20, 21]
19 out: [3, 5, 9, 10, 11, 12, 19, 20, 21]
20 in:  [3, 5, 9, 10, 11, 12, 14, 17, 18]
20 out: [3, 5, 9, 10, 11, 12, 17, 18, 20]
21 in:  [3, 5, 9, 10, 11, 12, 17, 18, 20]
21 out: [3, 5, 9, 10, 11, 12, 17, 20, 21]
22 in:  [3, 5, 9, 10, 11, 12, 19, 20, 23, 24]
22 out: [3, 5, 9, 10, 11, 12, 20, 22, 23, 24]
23 in:  [3, 5, 9, 10, 11, 12, 19, 20, 21]
23 out: [3, 5, 9, 10, 11, 12, 19, 20, 21, 23]
24 in:  [3, 5, 9, 10, 11, 12, 19, 20, 21, 23]
24 out: [3, 5, 9, 10, 11, 12, 19, 20, 23, 24]
25 in:  [3, 5, 9, 10, 11, 12, 22, 23, 26, 27]
25 out: [3, 5, 9, 10, 11, 12, 23, 25, 26, 27]
26 in:  [3, 5, 9, 10, 11, 12, 20, 22, 23, 24]
26 out: [3, 5, 9, 10, 11, 12, 22, 23, 24, 26]
27 in:  [3, 5, 9, 10, 11, 12, 22, 23, 24, 26]
27 out: [3, 5, 9, 10, 11, 12, 22, 23, 26, 27]
28 in:  [3, 5, 9, 10, 11, 12, 23, 25, 26, 29]
28 out: [3, 5, 9, 10, 11, 12, 23, 26, 28, 29]
29 in:  [3, 5, 9, 10, 11, 12, 23, 25, 26, 27]
29 out: [3, 5, 9, 10, 11, 12, 23, 25, 26, 29]
30 in:  [3, 5, 9, 10, 11, 12, 23, 26, 28, 31]
30 out: [3, 5, 9, 10, 11, 12, 23, 26, 30, 31]
31 in:  [3, 5, 9, 10, 11, 12, 23, 26, 28, 29]
31 out: [3, 5, 9, 10, 11, 12, 23, 26, 28, 31]
32 in:  [3, 5, 9, 10, 11, 12, 23, 26, 30, 33]
32 out: [3, 5, 9, 10, 11, 12, 23, 26, 30, 33]
33 in:  [3, 5, 9, 10, 11, 12, 23, 26, 30, 31]
33 out: [3, 5, 9, 10, 11, 12, 23, 26, 30, 33]
34 in:  [3, 5, 9, 10, 11, 12, 23, 26, 30, 33]
34 out: [3, 5, 9, 10, 11, 12, 23, 26, 30, 33]
exit: [3, 5, 9, 10, 11, 12, 13, 14, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 33]
Method: <init>
entry: []
1 in:  [2]
1 out: [2]
2 in:  []
2 out: [2]
3 in:  [2]
3 out: [2]
exit: [2]
